velox/functions/prestosql/window/tests/SimpleAggregatesTest.cpp

```diff
@@ -109,41 +143,15 @@ TEST_P(MultiAggregatesTest, randomInput) {
   testWindowFunction(vectors, overClauses);
 }
 
-TEST_P(MultiAggregatesTest, basicRangeFrames) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeBasicVectors(50)}, kFrameOverClauses, kRangeFrameClauses);
-}
-
-TEST_P(MultiAggregatesTest, basicRangeFramesWithSortOrders) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeBasicVectors(50)}, kSortOrderBasedOverClauses, kRangeFrameClauses);
-}
-
-TEST_P(MultiAggregatesTest, singlePartitionRangeFrames) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeSinglePartitionVector(100)}, kFrameOverClauses, kRangeFrameClauses);
-}
-
-TEST_P(MultiAggregatesTest, basicRowFrames) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeBasicVectors(50)}, kFrameOverClauses, kRowsFrameClauses);
-}
-
-TEST_P(MultiAggregatesTest, singlePartitionRowFrames) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeSinglePartitionVector(100)}, kFrameOverClauses, kRowsFrameClauses);
-}
+VELOX_INSTANTIATE_TEST_SUITE_P(
+    SimpleAggregatesTest,
+    MultiFrameAggregatesTest,
+    testing::ValuesIn(getTestParams(kFrameClauses)));
 
 VELOX_INSTANTIATE_TEST_SUITE_P(
     SimpleAggregatesTest,
     MultiAggregatesTest,
-    testing::ValuesIn(
-        {std::string("sum(c2)"),
-         std::string("min(c2)"),
-         std::string("max(c2)"),
-         std::string("count(c2)"),
-         std::string("avg(c2)"),
-         std::string("sum(1)")}));
+    testing::ValuesIn(getTestParams({"range unbounded preceding"})));
```

Comment:
Why do we use only one frame here and not kFrameClauses ?

Reply:
I understand that overClauses might be different. But do we need frameClauses to be different also ? why ? It seems restrictive to try random input only with range unbounded preceding.

Reply:
Thanks for the suggestion, it would be better to test random input for all frame clauses but the following changes would be needed:
1. overClauses to be modified 
2. kFrameClauses list would need to be changed because frames like "c2 preceding and current row" cannot be used with random fuzzer output since the column can contain negative values.  
3. Another class would be needed, parameterized with the frame clauses from the new frameClauses list (since instantiating the same class more than once is not allowed).
Please let me know if I should change accordingly.

Reply:
So the purpose of the random test data was to test partitioning and ordering for random distribution of data. We could change the randomInput() test data so that c1 and c2 have values that are always >0. And then have additional columns say c4, c5 with random values. c4 and c5 replace the current c1 and c2. The overClauses can be changed to use the newly added columns. Then that test can be used with all window frames. Does that work ?

Reply:
Thanks, updated accordingly.